### PR TITLE
Addition of SmoothMix Training in the Randomized Smoothing Module

### PR DESCRIPTION
**Pull request by asarj**
# Description

- Added support for SmoothMix training on Randomized Smoothing classifiers. 
- Added a `train_method` argument in the `fit` method via kwargs with support for SmoothMix training. 
- Added additional params in the `RandomizedSmoothingMixin` and `PyTorchRandomizedSmoothing` modules with the parameters needed to train the classifier properly.
- Implemented a `fit_pytorch` method that performs the SmoothMix training via PyTorch

Paper Link: https://arxiv.org/pdf/2111.09277.pdf
Code Repository based on paper: https://github.com/jh-jeong/smoothmix

Fixes # (issue)

## Type of change

Please check all relevant options.

- [X] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [ ] Test A
- [ ] Test B

**Test Configuration**:
- OS: Ubuntu 22.04 LTS
- Python version: 3.7
- ART version or commit number: 1.10.0
- TensorFlow / Keras / PyTorch / MXNet version: PyTorch 1.11.0+cu113

# Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by beat-buesser** [comment]
Hi @asarj Thank you very much for this draft pull request. I'll take look soon to follow your work. Please let us know if you have questions, etc.!

**Comment by asarj** [comment]
Thanks @beat-buesser! Tagging @keykholt and @rahmati so that they follow along with updates as well

**Comment by beat-buesser** [comment]
Hi @asarj It looks like some commits unrelated to this PR got signed by the recent force push. Could you please use `git cherry-pick` to move your commits onto a clean local version of `dev_1.11.0` and push it to your remote branch `smoothmix`?

**Comment by asarj** [comment]
Thanks for the heads up @beat-buesser, I've never used `git cherry-pick` before, but I think I addressed mistakenly signed commits by switching to the `dev_1.11.0` branch, `cherry-pick`ing my commits specifically, and pushing that back to the `smoothmix` branch. Can you please take a look and let me know if this has been resolved?

**Comment by asarj** [comment]
Also @beat-buesser, do I need to address anything from the lgtm-com bot's recent comment on fixing one alert?

File: art/estimators/certification/randomized_smoothing/pytorch.py [hunk 1]
```diff
@@ -26,15 +26,17 @@
 from typing import List, Optional, Tuple, Union, TYPE_CHECKING
 
 import numpy as np
+import torch
```
Comment (keykholt) [inline art/estimators/certification/randomized_smoothing/pytorch.py:29]: Should move this back to line 36

Comment (asarj) [reply art/estimators/certification/randomized_smoothing/pytorch.py]: Done and resolving this

**Comment by beat-buesser** [comment]
Hi @asarj Can you check the commits again, it looks like there are some unrelated commits left from May 8.

Can we implement `SmoothMixPGD` in `art.attacks.evasion` inheriting from  `art.atttacks.attack.EvasionAttack`?

Can we move the training procedure in to the fit method of the randomised smoothing estimator?

Should we create a separate estimator for SmoothMix? I think SmoothMix is adding a large umber of arguments to the randomised smoothing estimator which might make it more challenging to recognise which arguments are required for randomised smoothing.

**Comment by asarj** [comment]
@beat-buesser I can create a separate estimator for smoothmix and model this based on the randomized smoothing estimator and look into implementing SmoothMixPGD inheriting from `art.atttacks.attack.EvasionAttack` but this will take some time

**Comment by asarj** [comment]
Closing this because the changes have undergone a refactor based on recent feedback and are in a new PR found here: https://github.com/Trusted-AI/adversarial-robustness-toolbox/pull/1705#issuecomment-1131165716.

**Pull request closed without merging**